lib/galaxy/jobs/handler.py

```diff
@@ -558,7 +558,14 @@ def __verify_job_ready(self, job, job_wrapper):
 
         if state == JOB_READY:
             state = self.__check_user_jobs(job, job_wrapper)
-        if state == JOB_READY and self.app.config.enable_quotas:
+        # If user has plugged a media, then they might have enough quota
+        # on their media; hence, we should not raise the "over quota" flag
+        # checking the default storage only. If their usage exceeds their
+        # total quota on all their media, ObjectStore raises appropriate
+        # exception(s).
+        if state == JOB_READY and self.app.config.enable_quotas and \
+                (job.user is not None and
+                 (job.user.active_storage_media is None or not job.user.has_active_storage_media())):
```

Comment:
I think this should be redone on top of https://github.com/galaxyproject/galaxy/pull/10221 - which I think abstracts out the quota checking into nice optimizable functions. I think rather than checking if has_active_storage_media we should build on the abstractions in that PR to just ask if the configured objectstore we're talking to has quota left and then we can disable quota on objectstores that use storage media. The ability to disable quota on an objectstore is included in that PR.